near/src/runtime.rs

```diff
@@ -582,10 +582,10 @@ mod test {
             )]
         );
         let account = nightshade.view_account(state_root, &block_producers[0].account_id).unwrap();
-        assert_eq!(
-            (account.amount, account.staked),
-            (TESTING_INIT_BALANCE - TESTING_INIT_STAKE * 2, 2 * TESTING_INIT_STAKE)
-        );
+        // This test cannot account for all fees therefore we test that staked amount was staked
```

Comment:
why?

Reply:
It explains why.

Reply:
I don't understand why it cannot account for all fees.

Reply:
These tests have problems in general, please read the attached blog posts:

* It is not clear what the unit test is actually testing. It checks for all kinds of values, including nonces, hashes, balances, etc;
* We should have few integration tests and even fewer end-to-end tests, because they require a lot of maintenance -- a lot of changes to the domain logic require modifications to integration tests and end-to-end tests. These unit tests here are actually integration tests in disguise;
* These unit tests are too large and check everything, see: https://testing.googleblog.com/2018/06/testing-on-toilet-keep-tests-focused.html
* If we want to test the side effect of staking on the balance we should be checking it in the module responsible for staking, e.g. `ValidatorManager `. The side effect of rent should be checked in module that applies rent. The fees should be checked in near-vm-runtime.
* Some fees (like storage rent) have non-deterministic nature. We cannot assert a specific balance, because it depends on a multitude of external factors, like how many blocks have passed;
* Threading such information will violate encapsulation. `near/runtime.rs` does not have an access to the economics config at the moment, because it is encapsulated in the runtime crates. Exposing such config would break encapsulation.